pumps/splunk.go

```diff
@@ -145,28 +147,72 @@ func (p *SplunkPump) WriteData(ctx context.Context, data []interface{}) error {
 	for _, v := range data {
 		decoded := v.(analytics.AnalyticsRecord)
 		apiKey := decoded.APIKey
-		if p.config.ObfuscateAPIKeys {
-			if len(apiKey) > 4 {
-				apiKey = "****" + apiKey[len(apiKey)-4:]
-			}
+
+		// Check if the APIKey obfuscation is configured and its doable
+		if p.config.ObfuscateAPIKeys && len(apiKey) > p.config.ObfuscateAPIKeysLength {
+			// Obfuscate the APIKey, starting with 4 asterics and followed by last N chars (configured separately) of the APIKey
+			// The default value of the length is 0 so unless another number is configured, the APIKey will be fully hidden
+			apiKey = "****" + apiKey[len(apiKey)-p.config.ObfuscateAPIKeysLength:]
```

Comment:
Why do you need to obfuscate this field? with auth token this field supposed to have hashed value (unless you set the GW to not hash keys)